serverless/serverless
# Add a bucket to the provider for deployments.

## What did you implement:

***Implementing Issue:*** #1513 



## How did you implement it:

Added a check for `deploymentBucket ` on the provider object.  If it is there, then remove the S3 resource from the CloudFormation setup.  Also all deploys are now prefixed with `serverless/${serviceName}/` so multiple deploys can happen without deleting other services.


## How can we verify it:
Use a config similar to this: 
```json
service: pr1513
provider:
  name: aws
  runtime: nodejs4.3
  stage: test
  region: us-west-2
  deploymentBucket: com.serverless.deploys
```
After the deploy you will see the S3 resource created initially, then removed from the CF file and the bucket created and deleted, finally in `com.serverless.deploys` there will be a new folder `serverless/pr1513/${dateString}/` that contains the deployment files.




## Todos:


Also prefix deployments so they can co-exist with other files
